src/cpdb/helpers.py

```diff
@@ -28,44 +23,16 @@
 """a feedback from the group is to have a dictionary from the abbreviation for agency for something more explicity
 where would this list comes from? """
 
-def s3_resource():
-    return boto3.resource(
-        "s3",
-        aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
-        aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
-        endpoint_url=os.getenv("AWS_S3_ENDPOINT"),
-    )
-
-def get_csv(bucket, csv_filename):
-    obj = s3_resource().Object(bucket_name=bucket, key=csv_filename)
-    s = str(obj.get()["Body"].read(), "utf8")
-    data = StringIO(s)
-    return pd.read_csv(data, encoding="utf8")
-
-def zip_from_DO(zip_filename, bucket):
-    zip_obj = s3_resource().Object(bucket_name=bucket, key=zip_filename)
-    buffer = BytesIO(zip_obj.get()["Body"].read())
-
-    return ZipFile(buffer)
-
-def unzip_shapefile(zipfile, table):
-    try:
-        with zipfile as zf:
-            time = str(datetime.now().timestamp)
-            zf.extractall(path=f".library/{time}/{table}/")
-            gdf = gpd.read_file(f".library/{time}/{table}/{table}.shp")
-            shutil.rmtree(path=f".library/{time}")
-            return gdf
-    except:
-        return None
 
 def get_geometries(branch, table) -> dict:
-    points_zip = zip_from_DO(
-        zip_filename=f"db-cpdb/{branch}/latest/output/{table}.shp.zip",
-        bucket=BUCKET_NAME,
+    client = digital_ocean_client()
+
+    points_zip = client.zip_from_DO(
+        zip_filename=f"db-cpdb/{branch}/latest/output/{table}.shp.zip"
     )
-    gdf = unzip_shapefile(
-        zipfile=points_zip, table=table,
+    gdf = client.unzip_shapefile(
+        zipfile=points_zip,
+        table=table,
```

Comment:
Geopandas can take a buffer object and knows what to do with it even if it's zipped. I changed `return ZipFile(buffer)` to `return buffer` in the `zip_from_DO` method and then passed that straight to geopanda's `read_file` and it seemed to work. No need to extract to a local file and then remove it. Unless I'm missing something here?

Reply:
No you might be right - to be clear this isn't new code, but if that's possible that's great. Think Te and I had been hung up on the fact that a shapefile is really a folder of files last time, but if you've got a solution would be great to switch to it.